### ISSUE TITLE
Bootstrap tooltip conflict

### ISSUE THREAD

**Issue by annayafi** [mistic100/Bootstrap-Confirmation#63]
It would be great if we could have the tooltip bootstrap feature working with the confirmation feature along. I mean, when you enable confirmation on an element, its title doesn't display anymore, with "data-title" attribute. I understand it's better because the title would otherwise hide the confirmation popover. But what if you could see the bootstrap tooltip and when you click, this one disappears and let the confirmation display ?
With this code,
` <button data-toggle="confirmation" title="Êtes-vous sûr ?"/>`
` $('[data-toggle=confirmation]').confirmation();`

 I have:
![tooltip conflict](https://user-images.githubusercontent.com/8353486/28138509-50a92b5c-6751-11e7-8a07-f97235470fee.gif)

I added this as a script (typescript) :
 $('[data-toggle=confirmation]').parent("div").on("DOMNodeInserted",
            (a) => {
                if ($(a.target).hasClass("confirmation")) {
                    $(".tooltip").tooltip("hide");
                }
            });
And I get what I want:
![tooltip_noconflict](https://user-images.githubusercontent.com/8353486/28138702-ffbf437e-6751-11e7-8b2d-28b11e5ddf8a.gif)

Maybe it could be made by design, testing if there's any tooltip associated with some element ? This is just a suggestion or for any other users, a tip !

**Comment by mistic100** [comment]
Though I honestly think your example is not legit at all (the tooltip does not add any useful information, "Supprimer l'utilisateur", or whatever, would be better) I'll think about it, but it's very minor.

**Comment by annayafi** [comment]
Yes indeed I agree it's very minor, I just wanted to document it somewhere for others (maybe I got the wrong location), but it's not a real issue of course. 
This is just an example here, not the real business messages though. 
But the fact remains that some users can call me saying "I'm afraid to click on it because there is no messages (they mean title) like all the other sensitive buttons when I hover over them, and I'm not sure of what it's gonna do". Some buttons may do some work you can't rollback and then need extra explanations before you go for it and I can understand that they don't want to click on it before being sure (nothing tells them the explanation will be in the confirmation popup). Anyway, I had to tell them personally that they need to click on it to see the explanations. With the confirmation and the title, they feel reassured. But I didn't mean it as a real issue.

**Comment by mistic100** [comment]
Please open another issue if it is not the same topic.

If read carefully the doc http://bootstrap-confirmation.js.org/#events you will see there are the "show.bs.confirmation" and "hide.bs.confirmation" events.

**Comment by seyfer** [comment]
@mistic100 do we have different versions of the doc? see what I see
![screenshot from 2018-10-02 20-06-04](https://user-images.githubusercontent.com/1483925/46367730-adde3500-c67e-11e8-995b-926ea9b04b91.png)

they are not described and callbacks are not described.

and according to your question about if it is related - it is.
instead of a custom event, as @annayafi did above, it would be nice to put the code in the `onShow` callback.
But it is not in the docs and not working, I have tried. Looking forward to having it.

```
onShow: function() {
	$(this).parent("div").on("DOMNodeInserted",
            (a) => {
                if ($(a.target).hasClass("confirmation")) {
                    $(".tooltip").tooltip("hide");
                }
            });
}
```

**Comment by mistic100** [comment]
Ok, your first message seemed not related at all.

Now what can't you use events ?

```js
$('[data-toggle=confirmation]').on('show.bs.confirmation', function() {
  $(this).tooltip('hide');
});
```

Now to be clear, I won't hide callbacks for something that already works with events.

**Comment by seyfer** [comment]
@mistic100 thank you for the answer. :smiley: 
in my opinion, it is easier with the callback to make it more specific to one specific element. as we specify `selector` in the options. maybe I can use the same selector for attaching the event. will try.

great lib, keep moving :+1:
